src/web/components/SharingPermission/SharingPermissionsTable.tsx

```diff
@@ -0,0 +1,86 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useState } from 'react';
+
+import { ParticipantPayload } from '../../services/participant';
+import { ParticipantsTable } from './ParticipantsTable';
+
+import './SharingPermissionsTable.scss';
+
+type SharingPermissionsTableProps = {
+  sharedParticipants: ParticipantPayload[];
+};
+
+function NoParticipant() {
+  return (
+    <div className='no-participants-container'>
+      <img src='/group-icon.svg' alt='group-icon' />
+      <div className='no-participants-text'>
+        <h1>No Participants</h1>
+        <span>You don&apos;t have any sharing permissions yet.</span>
+      </div>
+    </div>
+  );
+}
+export function SharingPermissionsTable({ sharedParticipants }: SharingPermissionsTableProps) {
+  const [filter, setFilter] = useState('');
+  const [checkedParticipants, setCheckedParticipants] = useState<number[]>([]);
+  const [selectAll, setSelectAll] = useState(false);
+
+  const handleSelectAllChange = () => {
+    setSelectAll(!selectAll);
+    if (selectAll) {
+      setCheckedParticipants([]);
```

Comment:
shouldn't this condition be reversed ? check all if selectAll and viceversa ?

Reply:
@Kshitij-Banerjee when the handleSelectAllChange is called, the selectAll is not updated yet. I agree this looks unclear. I will change this to use an useEffect